api/v1alpha1/fault_injection.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// FaultInjection defines the fault injection policy to be applied. Support delays and aborts.
+type FaultInjection struct {
+
+	// If specified, the delay will inject a fixed delay into the request
+	//
+	// +optional
+	Delay *DelayConfig `json:"delay,omitempty"`
+
+	// If specified, the abort will abort the request with the specified HTTP status code
+	//
+	// +optional
+	Abort *AbortConfig `json:"abort,omitempty"`
+}
+
+// DelayConfig defines the delay fault injection configuration
+type DelayConfig struct {
+	// FixedDelay specifies the fixed delay duration
+	//
+	// +required
+	FixedDelay *metav1.Duration `json:"fixedDelay"`
+
+	// Percentage specifies the percentage of requests to be delayed. Default 100%, if set 0, no requests will be delayed.
```

Comment:
so we can only express int e.g. 1%, 2%, but not fractions such as 0.5% ?

Reply:
update